Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift

```diff
@@ -43,18 +43,80 @@ public extension PaywallComponent {
         public let heicLowRes: URL
     }
 
-    struct ColorInfo: Codable, Sendable, Hashable, Equatable {
+    struct ColorScheme: Codable, Sendable, Hashable, Equatable {
 
-        public init(light: ColorHex, dark: ColorHex? = nil) {
+        public init(light: ColorInfo, dark: ColorInfo? = nil) {
             self.light = light
             self.dark = dark
         }
 
-        public let light: ColorHex
-        public let dark: ColorHex?
+        public let light: ColorInfo
+        public let dark: ColorInfo?
 
     }
 
+    enum ColorInfo: Codable, Sendable, Hashable {
+
+        case hex(ColorHex)
+        case alias(String)
+
+        public func encode(to encoder: any Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+
+            switch self {
+            case .hex(let hex):
+                try container.encode(ColorInfoTypes.hex.rawValue, forKey: .type)
+                try container.encode(hex, forKey: .value)
+            case .alias(let alias):
+                try container.encode(ColorInfoTypes.alias.rawValue, forKey: .type)
+                try container.encode(alias, forKey: .value)
+            }
+        }
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            let type = try container.decode(ColorInfoTypes.self, forKey: .type)
+
+            switch type {
+            case .hex:
+                let value = try container.decode(ColorHex.self, forKey: .value)
+                self = .hex(value)
+            case .alias:
+                let value = try container.decode(String.self, forKey: .value)
+                self = .alias(value)
+            }
+        }
+
+        // swiftlint:disable:next nesting
+        private enum CodingKeys: String, CodingKey {
+
+            case type
+            case value
+
+        }
+
+        // swiftlint:disable:next nesting
+        private enum ColorInfoTypes: String, Decodable {
+
+            case hex
+            case alias
+
+        }
+
+    }
+
+//    struct ColorInfo: Codable, Sendable, Hashable, Equatable {
```

Comment:
Nit: remove commented code

Reply:
Ha! I did notice this and it gets removed like 2 PRs in 😛